### PR TITLE
Cardano support

### PR DESCRIPTION
**Pull request by Defcon023**
Closes  - 
#201 
#202 
#203 
#204 
#208 
#211

### CODE REVIEW

File: script/config/networks.json [hunk 1]
```diff
@@ -20,14 +20,14 @@
     "setterFacet": "0xE175513929084AbeC6a1891dA28d4Ff8e7321f90"
   },
   "fuji": {
-    "assetReserves": "0xb574419eDF8AB976A3F72723E58Aa57c90fF7a7c",
-    "bridgeFacet": "0x0883871d950ab5570010b840CD5c512969F66024",
+    "assetReserves": "0x9d283e00D371E7f9C76858889e72B17484C5CB0d",
```
Comment (Defcon023) [inline script/config/networks.json:23]: @serejke Khalani protocol Instance addresses for Cardano testing

File: src/InterchainMessaging/facets/RequestProcessor/DefaultRequestProcessor.sol [hunk 2]
```diff
@@ -3,7 +3,16 @@ pragma solidity ^0.8.0;
 import "./AbstractRequestProcessor.sol";
 import "../../interfaces/IMessageReceiver.sol";
 
-contract DefaultRequestProcessor is RemoteStorage, AbstractRequestProcessor{
```
Comment (serejke) [inline src/InterchainMessaging/facets/RequestProcessor/DefaultRequestProcessor.sol:6]: Hm, what is the reason for this change?

`AbstractRequestProcessor` is now unused.

But `HyperlaneAdapter` tries to cast `nexus` to `IRequestProcessorFacet`:
https://github.com/tvl-labs/khalani-core/blob/6094f5eeef23852dbe437aa3d1180da86a72c4a9/src/InterchainMessaging/adapters/HyperlaneAdapter.sol#L67

File: src/LiquidityReserves/khalani/LiquidityProjector.sol [hunk 3]
```diff
@@ -88,33 +111,35 @@ contract LiquidityProjector is ILiquidityProjector {
...
         emit MintOrUnlock(chainId, target, tokens);
+        mirrorTokens = new Token[](tokens.length);
         for(uint i; i<tokens.length;){
             //call reserves manager to unlock kai / mint mirror tokens
-            address token = mirrorTokenMap[chainId][tokens[i].tokenAddress];
-            if(token == address(0)){
-                revert AssetNotFound(tokens[i].tokenAddress);
+            address tokenToMint = mirrorTokenMap[chainId][tokens[i].tokenAddress];
+            address tokenToUnlock = exportTokenMap[chainId][tokens[i].tokenAddress];
+            if(tokenToMint == address(0x0) && tokenToUnlock == address(0x0)){
```
Comment (serejke) [inline src/LiquidityReserves/khalani/LiquidityProjector.sol:122]: Maybe move this `else` statement to the end of `if-condition`? Will be consistent with `lockOrBurn`

Comment (Defcon023) [reply src/LiquidityReserves/khalani/LiquidityProjector.sol]: https://github.com/tvl-labs/khalani-core/pull/207/commits/57a6501f78fbc887833f1f79a03728f0a4b1de04

File: src/InterchainMessaging/facets/RequestProcessor/DefaultRequestProcessor.sol (hunk 2 above)
Comment (Defcon023) [reply src/InterchainMessaging/facets/RequestProcessor/DefaultRequestProcessor.sol:6]: https://github.com/tvl-labs/khalani-core/pull/207/commits/186946a99c987bbb08c53935ba12c612ba1f5162

File: src/LiquidityReserves/khalani/LiquidityProjector.sol [hunk 4]
```diff
@@ -88,33 +119,40 @@ contract LiquidityProjector is ILiquidityProjector {
...
-            }
-            tokens[i].tokenAddress = token;
-            if(token == address(kai)){
+            address tokenToMint = mirrorTokenMap[chainId][tokens[i].tokenAddress];
+            address tokenToUnlock = exportTokenMap[chainId][tokens[i].tokenAddress];
+
+            if(tokenToUnlock!=address(0x0)){
+                //upscale before unlocking
+                //upscale amount according to exported token's decimal
+                outTokens[i].amount = LibScaling._upscale(tokens[i].amount,
```
Comment (serejke) [inline src/LiquidityReserves/khalani/LiquidityProjector.sol:134]: Agreed to create a single function:
```
function rescale(address token, uint256 amount, uint targetDecimals) internal pure returns (uint256)
function rescale(uint tokenDecimals, uint256 amount, uint targetDecimals) internal pure returns (uint256)
```
that encapsulates the upscaling/downscaling maths.

**Comment by serejke** [comment]
One more task to be covered in this PR: https://github.com/tvl-labs/khalani-core/issues/211

File: src/LiquidityReserves/khalani/LiquidityProjector.sol (hunk 4 above)
Comment (Defcon023) [reply src/LiquidityReserves/khalani/LiquidityProjector.sol]: https://github.com/tvl-labs/khalani-core/pull/207/commits/abf975316158f7398058b15f63c1de689335dbb0

**Review by serejke — APPROVED** [review]
